Schema update for relational data (#1709)
finos/datahelix: docs/UserGuide.md

* For a larger profile example see [The schema documentation](user/Schema.md)
* Further examples can be found in [the Examples folder](https://github.com/finos/datahelix/tree/master/examples)

> A Beta feature has been developed for generating nested data (as a step towards support for relational data), for further information see [this](RelationalData.md) page.

Reviewer: ```suggestion
A Beta feature has been developed for generating nested data (as a step towards support for relational data), for further information see [the Relational data documentation](RelationalData.md).
```
